integration/client/snapshot_windows_test.go

```diff
@@ -0,0 +1,42 @@
+//go:build windows
+// +build windows
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package client
+
+import (
+	"testing"
+
+	winio "github.com/Microsoft/go-winio"
+
+	. "github.com/containerd/containerd"
+	"github.com/containerd/containerd/snapshots/testsuite"
+)
+
+func runTestSnapshotterClient(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	// The SeBackupPrivilege and SeRestorePrivilege gives us access to system files inside the container mount points
+	// (and everywhere on the system), without having to explicitly set DACLs on each location inside the mount point.
+	if err := winio.EnableProcessPrivileges([]string{winio.SeBackupPrivilege, winio.SeRestorePrivilege}); err != nil {
+		t.Fatal(err)
+	}
+	defer winio.DisableProcessPrivileges([]string{winio.SeBackupPrivilege, winio.SeRestorePrivilege})
```

Comment:
The tests inside `testsuite.SnapshotterSuite` run in parallel mode and I assume they will all complete before this defer executes. However I guess other tests could overlap and have an unexpected privilege in some runs but not others.

It's possible we can rejig this to take a per-thread privilege inside the relevant test routines, if we can rely on them staying on the same thread.

Otherwise, we could split the test suite by tag, I guess, so that tests that require backup/restore privileges are separated.

This is made slightly not-so-bad by the existing second integration test run with parallel disabled, so unless extra privileges cases a _failure_, a false-success should be caught on the second run.

Reply:
Let me try thread local privilege escalation.